### PR TITLE
Add docs for insights

### PR DESCRIPTION
**Pull request by shalabhc**
Insights is a new dagster cloud feature and this provides and overview as well as some guides for exporting metrics and integrating external metrics.

### CODE REVIEW

**Comment by shalabhc** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17053** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17053" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/17053?utm_source=stack-comment).

File: docs/content/dagster-cloud/insights.mdx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Insights is Dagster Cloud feature that provides visibility into historical usage and cost metrics such as run duration, credit usage and failures. This feature is available as a top level tab in the Dagster Cloud UI:
+
+<Image
+alt="Viewing the Insights tab in the Dagster UI"
+src="/images/dagster-cloud/insights/insights-tab.png"
+width={771}
+height={536}
+/>
+
+The Insights page shows a list of metrics in the left panel. For each metric the daily, weekly or monthly aggregated values are shown in a graph in the main panel. As of October 2023 the metrics are update once a day. The following metrics are available:
```
Comment (benpankow) [inline docs/content/dagster-cloud/insights.mdx:26]:
```suggestion
The Insights page shows a list of metrics in the left panel. For each metric the daily, weekly or monthly aggregated values are shown in a graph in the main panel. These metrics are updated available on a daily cadence. The following metrics are available:
```

phrasing nit/suggestion

File: docs/content/dagster-cloud/insights.mdx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+- **Dagster credits**: Dagster credit cost associated with computing this object. Dagster credits are charged for every step that is run, and for every asset that is materialized.
```
Comment (benpankow) [inline docs/content/dagster-cloud/insights.mdx:28]: we should link to the pricing page here
```suggestion
- **Dagster credits**: Dagster credit cost associated with computing this object. Dagster credits are charged for every step that is run, and for every asset that is materialized. For more information, [see the pricing FAQ](https://dagster.io/pricing#faq).
```

File: docs/content/dagster-cloud/insights.mdx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- **Compute duration**: Time spent computing steps. For jobs which run steps in parallel, the compute duration may be longer than the wall clock time it takes for the run to complete.
+- **Retry compute**: "Time spent computing steps, including time spent retrying failed steps. For jobs which run steps in parallel, the compute duration may be longer than the wall clock time it takes for the run to complete.
```
Comment (benpankow) [inline docs/content/dagster-cloud/insights.mdx:30]:
```suggestion
- **Retry compute**: Time spent computing steps, including time spent retrying failed steps. For jobs which run steps in parallel, the compute duration may be longer than the wall clock time it takes for the run to complete.
```

File: docs/content/dagster-cloud/insights/external-metrics.mdx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+from dagster_cloud.dagster_insights import (
+    create_snowflake_insights_asset_and_schedule,
+)
+
+snowflake_insights_definitions = create_snowflake_insights_asset_and_schedule(
+    date(2023, 10, 5),
+    allow_partial_partitions=True,
+    dry_run=False,
+    snowflake_resource_key="snowflake_insights",
+)
```
Comment (benpankow) [inline docs/content/dagster-cloud/insights/external-metrics.mdx:62]:
```suggestion
from dagster_snowflake import SnowflakeResource
from dagster import Definition, EnvVar

from dagster_cloud.dagster_insights import (
    create_snowflake_insights_asset_and_schedule,
)

snowflake_insights_definitions = create_snowflake_insights_asset_and_schedule(
    "2023-10-5",
    snowflake_resource_key="snowflake_insights",
)
```

File: docs/content/dagster-cloud/insights/external-metrics.mdx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+```python
+from dagster_snowflake import SnowflakeResource
+from dagster import Definition, EnvVar
+
+from dagster_cloud.dagster_insights import (
+    create_snowflake_insights_asset_and_schedule,
+)
+
+snowflake_insights_definitions = create_snowflake_insights_asset_and_schedule(
+    "2023-10-5",
```
Comment (benpankow) [inline docs/content/dagster-cloud/insights/external-metrics.mdx:57]:
```suggestion
    start_date="2023-10-5",
```
mix of args/kwargs is a bit odd

**Review by erinkcochran87 — COMMENTED** [review]
@shalabhc Back to you! We'll need to update the screenshots though - both of them are a bit fuzzy, and include the entire browser window (which is fine for review, but not release). LMK if you want help with this or for me to take a look in the AM.

I also moved the dbt + Snowflake instructions to their own guide. I did this b/c it seems we'll add support for other things later, and having everything on that one page will get overwhelming, quickly.

**Review by elementljarredc — APPROVED** [review]

**Pull request merged**